adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapter.java

```diff
@@ -96,4 +112,38 @@ protected String getTypeName() {
         }
         return result.future();
     }
+
+    private Future<ResourceIdentifier> mapMessage(final MqttContext ctx, final ResourceIdentifier targetAddress, final MqttPublishMessage message) {
+        return getRegistrationAssertion(targetAddress.getTenantId(), targetAddress.getResourceId(),
+                ctx.authenticatedDevice(), ctx.getTracingContext()).compose(registratonInfo -> {
+            final Promise<ResourceIdentifier> result = Promise.promise();
+            if (!registratonInfo.containsKey(RegistrationConstants.FIELD_MAPPER)) {
+                result.complete(targetAddress);
+                return result.future();
+            }
```

Comment:
The response to *get Registration Assertion* only contains the *default* properties to be included in all downstream messages. I don't think that this field should be configured as a *default* property, though ...

Reply:
Ok, this is valuable information to me. I was indeed not trying to get the _defaults_ property but trying to get the (newly created) mapper field as you see below. How do I get these properties?

Reply:
Let me try to understand what you want to do first. The device that has been authenticated at this point is a gateway, right? You now want to retrieve the configuration data that has been registered for the gateway in order to determine the URL to invoke for mapping payload data, right?

Reply:
yes, that's right.

Reply:
Ok, then you would probably want to access data that has been registered under `ext` for the gateway. However, the Device Registration API doesn't support this for now. We can, of course, change it to either support a dedicated *get Registration* operation or we also include the `ext` data in the response to the *get Registration Assertion* operation. We would need to make that optional, however, for the time being. Otherwise, we would introduce an implementation breaking change to the Device Registration API. This means that the whole mapping functionality that you want to add would only work with a device registry that supports returning the `ext` properties ...

Reply:
Why can't we get it the same as the VIA property retrieved in the following line:
https://github.com/eclipse/hono/blob/9a37b826a2dff6ed800c73966539ebb8cfb8eb1b/service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java#L1108

Reply:
I've changed the device registry GetAssertionPayload to provide it. Question remains: is this allowed?